A97-xds-jwt-call-creds.md

```diff
@@ -0,0 +1,210 @@
+
+When the credential is asked for a token for a data plane RPC, if the
+token is not yet cached or the cached token will expire within some
+fixed refresh interval (typically 1 minute), the credential will start
+re-reading it from the file.
+
+When a data plane RPC starts, if the token is cached and is not expired,
+the token will immediately be added to the RPC, and the RPC will
+continue. Otherwise (i.e., before the token is initially obtained or
+after the cached token has expired), the data plane RPC will be queued
+until the file read completes. When the file read completes, the
+result (either success or failure, as described below) will be applied
+to all queued data plane RPCs.
+
+Note that when the token's expiration time is less than the refresh
+interval in the future, a new data plane RPC being started will trigger
+a new file read, but the cached token value will still be used for
+that data plane RPC. This pre-emptive re-fetching is intended to avoid
+periodic latency spikes when refreshing the token.
+
+If reading the file fails, all queued data plane RPCs will be failed
+with UNAVAILABLE status.
+
+TODO: Use UNAUTHENTICATED instead of UNAVAILABLE if we know there was a
+permission problem with the file?
+
+If reading the file succeeds, the content of the file will contain the JWT
+token. which the client will cache. The client does not need to do full
+[RFC-7519](https://datatracker.ietf.org/doc/html/rfc7519) validation
+of the token (that is the responsibility of the server side), but it
+does need to extract the `exp` field for caching purposes. If the `exp`
+field cannot be extracted (i.e., the JWT token is invalid), all queued
+data plane RPCs will be failed with status UNAUTHENTICATED. Otherwise,
+the cache is updated, and the returned token is added to all queued data
+plane RPCs, which may then continue.
+
+If reading the file does not result in the cache being updated (i.e.,
+if reading the file fails or if it returns an invalid JWT token),
+[backoff](https://github.com/grpc/grpc/blob/master/doc/connection-backoff.md)
+must be applied before the next attempt may be started. If a data plane
+RPC is started when there is no cached token available and while in
+backoff delay, it will be failed with the status from the last attempt
+to read the file.  When the backoff delay expires, the next data plane
+RPC will trigger a new attempt. Note that no attempt should be started
+until and unless a data plane RPC is started, since we do not want to
+unnecessarily retry if the channel is idle. The backoff state will be
+reset once the file has been read successfully.
+
+To add the token to a data plane RPC, the call credential will add
+a header named `authorization`. The header value will be the string
+`Bearer ` (note trailing space) followed by the token value.
+
+### Configuring Call Credentials in xDS Bootstrap File
+
+We will add a new field to the xDS bootstrap representation of an xDS
+server:
+
+```json
```

Comment:
Suggestion: you can use [`json5`](https://json5.org/) format to avoid syntax highlight errors produced by standard json validation.

```suggestion
```json5
```